BasicRecordingApiKotlin/app/src/main/java/com/google/android/gms/fit/samples/basicrecordingapikotlin/MainActivity.kt

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (C) 2014 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gms.fit.samples.basicrecordingapikotlin
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Color
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.Settings
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.widget.TextViewCompat
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.fit.samples.common.logger.Log
+import com.google.android.gms.fit.samples.common.logger.LogView
+import com.google.android.gms.fit.samples.common.logger.LogWrapper
+import com.google.android.gms.fit.samples.common.logger.MessageOnlyLogFilter
+import com.google.android.gms.fitness.Fitness
+import com.google.android.gms.fitness.FitnessOptions
+import com.google.android.gms.fitness.data.DataType
+import com.google.android.material.snackbar.Snackbar
+
+const val TAG = "BasicRecordingApi"
+
+/**
+ * This sample demonstrates how to use the Recording API of the Google Fit platform to subscribe
+ * to data sources, query against existing subscriptions, and remove subscriptions. It also
+ * demonstrates how to authenticate a user with Google Play Services.
+ *
+ * Because the subscription is for data type TYPE_CALORIES_EXPENDED, on Android 10 the Android
+ * ACTIVITY_RECOGNITION permission is required. This sample shows how to check for and request the
+ * permission when necessary.
+ */
+class MainActivity : AppCompatActivity() {
+    private val fitnessOptions: FitnessOptions by lazy {
+        FitnessOptions.builder()
+                .addDataType(DataType.TYPE_CALORIES_EXPENDED, FitnessOptions.ACCESS_WRITE)
+                .build()
+    }
+
+    // Two maps are used in conjunction with the OAuth flow. This is in order to identify what
+    // function to execute once the flow has successfully completed.
+    private val signInActionMap = mapOf<() -> Any, Int>(
+            ::subscribe to 1,
```

Comment:
Constants with Request code in name so devs know right away the number assigned with it has more meaning.